lnwallet/channel.go

```diff
@@ -4801,6 +4801,19 @@ func (lc *LightningChannel) computeView(view *htlcView, remoteChain bool,
 	}
 	feePerKw := filteredHTLCView.feePerKw
 
+	// We need to first check ourBalance and theirBalance to be negative
+	// because MilliSathoshi is a unsigned type and can underflow in
+	// `evaluateHTLCView`. This should never happen for views which do not
+	// include new updates (remote or local).
+	if int64(ourBalance) < 0 {
```

Comment:
I just moved this check to the root func. imo, was prior also in `validateCommitmentSanity`. This check is important to prevent the Peer mess with its balance from our point of view.

https://github.com/lightningnetwork/lnd/blob/master/lnwallet/channel.go#L3869-L3881